### ISSUE TITLE
Structs are unweaveable

### ISSUE THREAD

**Issue by keith-anders** [vescon/MethodBoundaryAspect.Fody#37]
A weaved struct seems to throw exceptions.

```csharp
using static System.Console;

class Aspect : OnMethodBoundaryAspect
{
    public override void OnEntry(MethodExecutionArgs arg) => WriteLine("Arg was {0}", arg.Arguments[0]);
}

class Program
{
    static void Main(string[] args) => new Struct().DoSomething("Value");
}

public struct Struct
{
    [Aspect] public void DoSomething(string s) => WriteLine("Something: {0}", s);
}
```

I'll submit a PR in the next few days.

**Comment by marcells** [comment]
Cool, thank you. :+1: We did never try that.

**Comment by marcells** [comment]
Merged in pr #38.
